plugin/pkg/auth/authenticator/token/bootstrap/bootstrap.go

```diff
@@ -184,3 +192,28 @@ func parseToken(s string) (string, string, error) {
 	}
 	return split[1], split[2], nil
 }
+
+// getGroups loads and validates the bootstrapapi.BootstrapTokenGroupsKey key
+// from the bootstrap token secret, returning a list of group names or an error
+// if any of the group names are invalid.
+func getGroups(secret *api.Secret) ([]string, error) {
+	groupsString := getSecretString(secret, bootstrapapi.BootstrapTokenGroupsKey)
+	if groupsString == "" {
+		return []string{bootstrapapi.BootstrapDefaultGroup}, nil
+	}
+
+	// split on commas amd validate the prefix on each group name
+	groups := strings.Split(groupsString, ",")
+	for _, group := range groups {
+		if group == bootstrapapi.BootstrapDefaultGroup {
+			// allow the default group even though it doesn't strictly match the prefix
+			continue
+		}
+
+		if !strings.HasPrefix(group, bootstrapapi.BootstrapGroupPrefix) {
+			return nil, fmt.Errorf("group does not start with %q", bootstrapapi.BootstrapGroupPrefix)
```

Comment:
This gets wrapped by the `tokenError` helper above, so the actual warning message is:
`Bootstrap secret %s/%s matching bearer token has invalid value for key auth-groups: group does not start with "system:bootstrappers:".`